cidre/src/commonMain/kotlin/at/asitplus/cidre/IpAddressAndPrefix.kt

```diff
@@ -44,6 +45,39 @@ sealed interface IpAddressAndPrefix<N : Number, S : CidrNumber<S>> {
     /**`true` if this is (part of)  the [IpNetwork.SpecialRanges.multicast] network */
     val isMulticast: Boolean
 
+    /**
+     * Encodes this network into X.509 iPAddressName ByteArray (RFC 5280).
+     * IPv4: [4 bytes base address][4 bytes subnet mask]  (8 bytes)
+     * IPv6: [16 bytes base address][16 bytes subnet mask] (32 bytes)
+     */
+    fun toX509Octets(): ByteArray = address.octets + netmask
+
+    companion object {
+
+        /**
+         * Low-level helper used by [IpNetwork.fromX509Octets] and [IpInterface.fromX509Octets]
+         * to extract base address and CIDR prefix
+         * IPv4: [4 bytes base address][4 bytes subnet mask]  (8 bytes)
+         * IPv6: [16 bytes base address][16 bytes subnet mask] (32 bytes)
```

Comment:
```suggestion
         * * IPv4 byte layout: `AAAANNNN`, where `A` ist an address octet and `N` is a netmask octet (8 bytes total)
         * * IPv6 byte layout:  `AAAAAAAAAAAAAAAANNNNNNNNNNNNNNNN`, where `A` ist an address octet and `N` is a netmask octet(32 bytes total)
```

Reply:
Fee free to reformat, introduce spaces of whatever, if you feel it makes it more legible